src/google/protobuf/stubs/stringprintf_unittest.cc

```diff
@@ -147,6 +147,16 @@ TEST(StringPrintfTest, LargeBuf) {
   delete[] buf;
 }
 
+TEST(StringPrintfTest, HugeBuf) {
+  // Check that the a corner case buffer is handled correctly.
+  int n = std::numeric_limits<int>::max() ;
+  char* buf = (char *) malloc(n);
```

Comment:
Check whether buf is NULL before using it. We should make the test pass if the machine does not have 2G memory at all.